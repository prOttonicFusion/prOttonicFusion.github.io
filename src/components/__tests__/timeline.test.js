import React from 'react'
import { render } from '@testing-library/react'
import { jest } from '@jest/globals'
import Timeline from '../timeline'

jest.mock('../../sections/timeline.mdx', () => {
    return {
        __esModule: true,
        default: () => {
            return <div>Some content</div>
        },
    }
})

describe('Timeline', () => {
    it('renders timeline correctly', () => {
        const { container } = render(
            <Timeline />,
        )
        expect(container.firstChild).toMatchSnapshot()
    })
})
