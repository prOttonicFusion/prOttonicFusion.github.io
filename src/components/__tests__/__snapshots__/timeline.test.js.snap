// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders timeline correctly 1`] = `
<div
  class="section-container"
  id="timeline"
>
  <div>
    <div>
      Some content
    </div>
  </div>
  <div
    class="timeline-container"
  >
    <ul
      class="timeline"
    >
      <li
        class="timeline-item"
      >
        <div
          class="timeline-badge"
        >
          <i
            class="glyphicon glyphicon-check"
          />
        </div>
        <div
          class="timeline-panel"
        >
          <div
            class="timeline-heading"
          >
            <h4
              class="timeline-title"
            >
              Software Engineer
            </h4>
            <span
              class="timeline-detail-container"
            >
              <span
                class="timeline-detail-text timeline-place"
              >
                Koherent
              </span>
              <span
                class="timeline-detail-text timeline-spacer"
              >
                |
              </span>
              <span
                class="timeline-detail-text timeline-date"
              >
                2022 -
              </span>
            </span>
          </div>
          <div
            class="timeline-body"
          >
            <p>
              Full-stack web development
            </p>
          </div>
        </div>
      </li>
      <li
        class="timeline-item"
      >
        <div
          class="timeline-badge"
        >
          <i
            class="glyphicon glyphicon-check"
          />
        </div>
        <div
          class="timeline-panel"
        >
          <div
            class="timeline-heading"
          >
            <h4
              class="timeline-title"
            >
              Software Developer
            </h4>
            <span
              class="timeline-detail-container"
            >
              <span
                class="timeline-detail-text timeline-place"
              >
                Fjuul
              </span>
              <span
                class="timeline-detail-text timeline-spacer"
              >
                |
              </span>
              <span
                class="timeline-detail-text timeline-date"
              >
                2020 - 2022
              </span>
            </span>
          </div>
          <div
            class="timeline-body"
          >
            <p>
              Full-stack mobile & web development
            </p>
          </div>
        </div>
      </li>
      <li
        class="timeline-item"
      >
        <div
          class="timeline-badge"
        >
          <i
            class="glyphicon glyphicon-check"
          />
        </div>
        <div
          class="timeline-panel"
        >
          <div
            class="timeline-heading"
          >
            <h4
              class="timeline-title"
            >
              MSc in Computational Materials Physics
            </h4>
            <span
              class="timeline-detail-container"
            >
              <span
                class="timeline-detail-text timeline-place"
              >
                University of Helsinki
              </span>
              <span
                class="timeline-detail-text timeline-spacer"
              >
                |
              </span>
              <span
                class="timeline-detail-text timeline-date"
              >
                2020
              </span>
            </span>
          </div>
          <div
            class="timeline-body"
          >
            <p />
          </div>
        </div>
      </li>
      <li
        class="timeline-item"
      >
        <div
          class="timeline-badge"
        >
          <i
            class="glyphicon glyphicon-check"
          />
        </div>
        <div
          class="timeline-panel"
        >
          <div
            class="timeline-heading"
          >
            <h4
              class="timeline-title"
            >
              Research Assistant
            </h4>
            <span
              class="timeline-detail-container"
            >
              <span
                class="timeline-detail-text timeline-place"
              >
                Univ. of Helsinki Accelerator Lab
              </span>
              <span
                class="timeline-detail-text timeline-spacer"
              >
                |
              </span>
              <span
                class="timeline-detail-text timeline-date"
              >
                2017 - present
              </span>
            </span>
          </div>
          <div
            class="timeline-body"
          >
            <p>
              Molecular dynamics simulations related to fusion materials research
            </p>
          </div>
        </div>
      </li>
      <li
        class="timeline-item"
      >
        <div
          class="timeline-badge"
        >
          <i
            class="glyphicon glyphicon-check"
          />
        </div>
        <div
          class="timeline-panel"
        >
          <div
            class="timeline-heading"
          >
            <h4
              class="timeline-title"
            >
              BSc in Physics
            </h4>
            <span
              class="timeline-detail-container"
            >
              <span
                class="timeline-detail-text timeline-place"
              >
                University of Helsinki
              </span>
              <span
                class="timeline-detail-text timeline-spacer"
              >
                |
              </span>
              <span
                class="timeline-detail-text timeline-date"
              >
                2018
              </span>
            </span>
          </div>
          <div
            class="timeline-body"
          >
            <p />
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
